persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/stringbased/JdbcStringBasedStore.java

```diff
@@ -257,6 +258,52 @@ private void executeLegacyUpdate(Connection connection, MarshalledEntry entry, S
    }
 
    @Override
+   public void writeBatch(List<MarshalledEntry<? extends K, ? extends V>> marshalledEntries) {
+      // If upsert is not supported, then we must execute the legacy write for each entry; i.e. read then update/insert
+      if (!tableManager.isUpsertSupported()) {
+         marshalledEntries.forEach(this::write);
```

Comment:
If we call super, we have to call `AdvancedLoadWriteStore.super.writeBatch(marshalledEntries)` as it is a default interface implementation. So calling `marshalledEntries.forEach(this::write)` is more explicit and succinct.